pkg/pillar/cmd/zedrouter/networkinstance.go

```diff
@@ -95,23 +96,23 @@ func checkPortAvailable(
 		// Make sure it is configured for IP or will be
 		if portStatus.Dhcp == types.DT_NONE {
 			errStr := fmt.Sprintf("Port %s not configured for shared use. "+
-				"Cannot be used by non-Switch Network Instance %s-%s\n",
-				status.Port, status.UUID, status.DisplayName)
+				"Cannot be used by Switch Network Instance %s-%s\n",
+				status.CurrentUplinkIntf, status.UUID, status.DisplayName)
 			return errors.New(errStr)
 		}
 		// Make sure it is not used by a NetworkInstance of type Switch
 		for _, iterStatusEntry := range ctx.networkInstanceStatusMap {
```

Comment:
@naiming-zededa You use ctx.networkInstanceStatusMap for iteration, but are not updating it after making changes to status. Don't know where else we end up doing the same thing. I think it better to get all the NetworkInstanceStatus using pubsub and then work on them.